code/online_challenges/src/hackerrank/queens_attack_II/queens_attack_II.cpp

```diff
@@ -0,0 +1,55 @@
+#include <iostream>
+
+int main() 
+{
+  int n,k , rQ, cQ, rObj, cObj;
+  
+  std::cin >> n >> k >> rQ >> cQ;
+
+  //Directions available for queen to move
+  int N = n - rQ;
+  int S = rQ - 1;
+  int E = n - cQ;
+  int W = cQ - 1;
+  int NE = (n - rQ < n - cQ ? n - rQ : n - cQ);
+  int SE = (rQ - 1 < n - cQ ? rQ - 1 : n - cQ);
+  int NW = (n - rQ < cQ - 1 ? n - rQ : cQ - 1);
+  int SW = (rQ - 1 < cQ - 1 ? rQ - 1 : cQ - 1);
+
+  for (int i = 0; i < k; ++i)
+  {
+    std::cin >> rObj >> cObj;
+
+    if (cQ == cObj) {
+      if ((rObj > rQ) && (rObj - rQ - 1 < N))
+        N = rObj - rQ - 1; //N
+      else if ((rObj < rQ) && (rQ - rObj - 1 < S)) 
+        S = rQ - rObj-1; //S
+    }
+    else if (rQ == rObj) 
+    {
+      if ((cObj > cQ) && (cObj - cQ - 1 < E)) 
+        E = cObj - cQ - 1; //E        
+      else if ((cObj < cQ) && (cQ - cObj - 1 < W)) 
+        W = cQ - cObj - 1; //W
+    }    
+    else if (rObj - rQ == cObj - cQ) 
+    {
+      if ((rObj > rQ) && (rObj - rQ - 1 < NE)) 
+        NE = rObj - rQ - 1; //NE
+      else if ((rQ > rObj) && (rQ - rObj - 1 < SW)) 
+        SW = rQ - rObj - 1; //SW
+    } 
+    else if (abs(rObj - rQ) == abs(cObj - cQ)) 
+    {
+      if ((rObj > rQ) &&(rObj - rQ - 1 < NW)) 
+        NW = rObj - rQ - 1; //NW
+      else if ((rQ > rObj) && (rQ - rObj - 1 < SE)) 
+        SE = rQ - rObj - 1; //SE           
+    }
+  }
+
+  std::cout << N + S + E + W + NE + SE + NW + SW;
+
+  return 0;
+}
```

Comment:
Please add newlines to the end of all files.

Reply:
Sir, the newlines are in there wherever required!
If I add newlines at the end of all files, it will end up affecting the formatting of output.
Please tell me what to do @arnavb

Reply:
What do you mean it will affect formatting of output? Files should always have an extra empty line at the end according to UNIX conventions.